site/e2e/reporter.ts

```diff
@@ -1,85 +1,145 @@
-import fs from "fs";
+/* eslint-disable no-console -- Logging is sort of the whole point here */
+import * as fs from "fs/promises";
 import type {
   FullConfig,
   Suite,
   TestCase,
   TestResult,
   FullResult,
   Reporter,
+  TestError,
 } from "@playwright/test/reporter";
 import axios from "axios";
+import type { Writable } from "stream";
 
 class CoderReporter implements Reporter {
+  config: FullConfig | null = null;
+  testOutput = new Map<string, Array<[Writable, string]>>();
+  passedCount = 0;
+  failedTests: TestCase[] = [];
+  timedOutTests: TestCase[] = [];
+
   onBegin(config: FullConfig, suite: Suite) {
-    // eslint-disable-next-line no-console -- Helpful for debugging
-    console.log(`Starting the run with ${suite.allTests().length} tests`);
+    this.config = config;
+    console.log(`==> Running ${suite.allTests().length} tests`);
   }
 
   onTestBegin(test: TestCase) {
-    // eslint-disable-next-line no-console -- Helpful for debugging
-    console.log(`Starting test ${test.title}`);
+    this.testOutput.set(test.id, []);
+    console.log(`==> Starting test ${test.title}`);
   }
 
-  onStdOut(chunk: string, test: TestCase, _: TestResult): void {
-    // eslint-disable-next-line no-console -- Helpful for debugging
-    console.log(
-      `[stdout] [${test ? test.title : "unknown"}]: ${chunk.replace(
-        /\n$/g,
-        "",
-      )}`,
-    );
+  onStdOut(chunk: string, test?: TestCase, _?: TestResult): void {
+    if (!test) {
+      const preserve = this.config?.preserveOutput === "always";
+      if (preserve) {
+        console.log(`[stdout] ${chunk.replace(/\n$/g, "")}`);
+      }
+      return;
+    }
+    this.testOutput.get(test.id)!.push([process.stdout, chunk]);
   }
 
-  onStdErr(chunk: string, test: TestCase, _: TestResult): void {
-    // eslint-disable-next-line no-console -- Helpful for debugging
-    console.log(
-      `[stderr] [${test ? test.title : "unknown"}]: ${chunk.replace(
-        /\n$/g,
-        "",
-      )}`,
-    );
+  onStdErr(chunk: string, test?: TestCase, _?: TestResult): void {
```

Comment:
We might be losing here output from `coderd`, which can be really useful during backend debugging. In the legacy version, it is logged like this:

```
[stderr] [unknown]: [WebServer] 2023-10-17 06:54:03.412 [info]  provisionerd.runner: apply successful  job_id=89020454-1bd8-4eed-a0c9-eb12fa532a92  template_name=68ac3052  template_version=hardcore_moser9  workspace_build_id=465c8cad-3fcb-42fb-b9bb-56007b4a1db3  workspace_id=74006ba8-82a8-4693-b971-f3da61ad02e0  workspace_name=d93260bd  workspace_owner=admin  workspace_transition=start  resource_count=1  resources="[name:\"example\" type:\"echo\"]"  state_len=0
```

It looks like the new reporter simply skips these log records.

Reply:
that was sort of intentional. it adds a lot of noise when trying to debug a frontend failure, and since it's not tied to a specific test case it's hard to buffer and display later in a way that makes sense.

I also added some logic for `preserveOutput`, but I just read the docs and it turns out it does something entirely different from what I expected, and can't even be set from the command line like I hoped 😵‍💫

Reply:
example run, which now has 41 occurrences of the word "error" but succeeded https://github.com/coder/coder/actions/runs/6549912328/job/17787879014?pr=10304

Reply:
A good compromise would be only filtering out `audit_log` entries, which do not bring a lot of value. 👍 

I would leave coderd output as it is helpful with debugging the request flow or provisioner behavior, especially while dealing with flaky issues.

Reply:
if we can also ignore lines like `echo: recv done on Session  session_id= error=EOF` then I'll be happy. :)

Reply:
Agree 👍